src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -615,6 +622,412 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
+  char errbuf[512];
+
+  uint64_t disable_CPU   = 0;
+  uint64_t disable_GLIBC = 0;
+  // glibc: sysdeps/x86/get-isa-level.h:
+  // glibc: if (CPU_FEATURE_USABLE_P (cpu_features, CMOV)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, CX8)
+  // glibc:     && CPU_FEATURE_CPU_P (cpu_features, FPU)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, FXSR)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, MMX)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE2))
+  // glibc:     isa_level = GNU_PROPERTY_X86_ISA_1_BASELINE;
+  if ((_features & CPU_CMOV) &&
+      (_features & CPU_CX8) &&
+      // FPU is always present on i686+: (_features & CPU_FPU) &&
+      (_features & CPU_SSE2))
+    // These cannot be disabled by GLIBC_TUNABLES.
+    if (excessive_CPU & (CPU_FXSR | CPU_MMX | CPU_SSE)) {
+      assert(!(excessive_CPU & CPU_SSE2), "(_features & CPU_SSE2) cannot happen");
+      // FIXME: The choice should be based on glibc impact, not the feature age.
+      // CX8 is i586+, CMOV is i686+ 1995+, SSE2 is 2000+
+      excessive_CPU |= CPU_SSE2;
+    }
+    if ((_features & CPU_FXSR) &&
+        (_features & CPU_MMX) &&
+        (_features & CPU_SSE)) {
+      // glibc: if (CPU_FEATURE_USABLE_P (cpu_features, CMPXCHG16B)
+      // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, LAHF64_SAHF64)
+      // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, POPCNT)
+      // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE3)
+      // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSSE3)
+      // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE4_1)
+      // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE4_2))
+      // glibc:     isa_level |= GNU_PROPERTY_X86_ISA_1_V2;
+      if ((_features & CPU_POPCNT) &&
+          (_features & CPU_SSSE3) &&
+          (_features & CPU_SSE4_1) &&
+          (_features & CPU_SSE4_2)) {
+        if ((excessive_CPU & CPU_SSE3) ||
+            (excessive_GLIBC & (GLIBC_CMPXCHG16 | GLIBC_LAHFSAHF))) {
+          assert(!(excessive_CPU & CPU_SSE4_2), "(_features & CPU_SSE4_2) cannot happen");
+          // POPCNT is 2007+, SSSE3 is 2006+, SSE4_1 is 2007+, SSE4_2 is 2008+.
+          excessive_CPU |= CPU_SSE4_2;
+        }
+        if ((_features & CPU_SSE3) &&
+            (_glibc_features & GLIBC_CMPXCHG16) &&
+            (_glibc_features & GLIBC_LAHFSAHF)) {
+          // glibc: if (CPU_FEATURE_USABLE_P (cpu_features, AVX)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, AVX2)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, BMI1)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, BMI2)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, F16C)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, FMA)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, LZCNT)
+          // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, MOVBE)) 
```

Comment:
SKARA complians on this line

Reply:
Hmm, I still see an error: https://github.com/openjdk/crac/pull/41/checks?check_run_id=13518791480